src/iuse.cpp

```diff
@@ -5059,9 +5060,16 @@ static bool heat_item( player &p )
     item &target = heat->is_food_container() ? heat->contents.front() : *heat;
     p.mod_moves( -300 );
     add_msg( _( "You heat up the food." ) );
-    target.item_tags.insert( "HOT" );
-    target.active = true;
-    target.item_counter = 600; // sets the hot food flag for 60 minutes
+    if( target.item_tags.count( "FROZEN" ) ) {
+        target.item_tags.erase( "FROZEN" );
+        target.item_tags.insert( "HOT" );
+        target.active = true;
+        target.item_counter = 100; // prevents insta-freeze afterwards
+    } else {
+        target.item_tags.insert( "HOT" );
+        target.active = true;
+        target.item_counter = 600; // sets the hot food flag for 60 minutes
```

Comment:
Food remains hot for an hour?

Reply:
This is a legacy value. I haven't changed it. Should I?

Reply:
IMO it should be changed to 10-15 minutes.

Reply:
How about making it weight related? Same with defrost moves consumption. More mass, more energy "stored",  longer the time needed to heat/cool. For ex. heating a small sausage takes less time then defrosting a galon of wood's soup. Also this sausage will be cold sooner then this amount soup. I know that IRL its also volume dependant and environment temperature plays big role too, but using only weight would be 
sufficient aproximation IMHO.  Can't say at this moment if its possible to do so, but will look at it asap.

Reply:
Sounds good.

Reply:
Done some research and found some interesting data that I have to share: [THIS](http://stanwagon.com/public/evaporationportmannwagonmier.pdf) paper on Newton Law of Cooling says that boiling water (100C) in room temperature drops to 37C (human body temp) in about 43 minutes, but excluding evaporation (adding oil film to the water) it drops to 42C in 60 minutes. Law scales with delta temperature and other variables, but I'm sane enough to don't even think of using the real math for this. However as for cooling from hot to normal, I now think that this 60 minutes was not that off, considering just the Newton's Law.

Reply:
Still, as [THIS](https://www.quora.com/Do-things-that-are-smaller-lose-heat-faster) great explanation says heat capacity plays huge role, so smaller objects lose temperature faster then larger ones. I have an idea of implementation now, and will try to present some results soon, as my free time allows me.